test/parallel/test-url-pathtofileurl.js

```diff
@@ -60,112 +61,118 @@ const url = require('url');
+  // equals
+  { path: 'C:\\foo=bar', expected: 'file:///C:/foo=bar' },
+  // colon
+  { path: 'C:\\foo:bar', expected: 'file:///C:/foo:bar' },
+  // semicolon
+  { path: 'C:\\foo;bar', expected: 'file:///C:/foo;bar' },
+  // percent
+  { path: 'C:\\foo%bar', expected: 'file:///C:/foo%25bar' },
+  // backslash
+  { path: 'C:\\foo\\bar', expected: 'file:///C:/foo/bar' },
+  // backspace
+  { path: 'C:\\foo\bbar', expected: 'file:///C:/foo%08bar' },
+  // tab
+  { path: 'C:\\foo\tbar', expected: 'file:///C:/foo%09bar' },
+  // newline
+  { path: 'C:\\foo\nbar', expected: 'file:///C:/foo%0Abar' },
+  // carriage return
+  { path: 'C:\\foo\rbar', expected: 'file:///C:/foo%0Dbar' },
+  // latin1
+  { path: 'C:\\fóóbàr', expected: 'file:///C:/f%C3%B3%C3%B3b%C3%A0r' },
+  // Euro sign (BMP code point)
+  { path: 'C:\\€', expected: 'file:///C:/%E2%82%AC' },
+  // Rocket emoji (non-BMP code point)
+  { path: 'C:\\🚀', expected: 'file:///C:/%F0%9F%9A%80' },
+  // UNC path (see https://docs.microsoft.com/en-us/archive/blogs/ie/file-uris-in-windows)
+  { path: '\\\\nas\\My Docs\\File.doc', expected: 'file://nas/My%20Docs/File.doc' },
+];
+const posixTestCases = [
+  // Lowercase ascii alpha
+  { path: '/foo', expected: 'file:///foo' },
+  // Uppercase ascii alpha
+  { path: '/FOO', expected: 'file:///FOO' },
+  // dir
+  { path: '/dir/foo', expected: 'file:///dir/foo' },
+  // trailing separator
+  { path: '/dir/', expected: 'file:///dir/' },
+  // dot
+  { path: '/foo.mjs', expected: 'file:///foo.mjs' },
+  // space
+  { path: '/foo bar', expected: 'file:///foo%20bar' },
+  // question mark
+  { path: '/foo?bar', expected: 'file:///foo%3Fbar' },
+  // number sign
+  { path: '/foo#bar', expected: 'file:///foo%23bar' },
+  // ampersand
+  { path: '/foo&bar', expected: 'file:///foo&bar' },
+  // equals
+  { path: '/foo=bar', expected: 'file:///foo=bar' },
+  // colon
+  { path: '/foo:bar', expected: 'file:///foo:bar' },
+  // semicolon
+  { path: '/foo;bar', expected: 'file:///foo;bar' },
+  // percent
+  { path: '/foo%bar', expected: 'file:///foo%25bar' },
+  // backslash
+  { path: '/foo\\bar', expected: 'file:///foo%5Cbar' },
+  // backspace
+  { path: '/foo\bbar', expected: 'file:///foo%08bar' },
+  // tab
+  { path: '/foo\tbar', expected: 'file:///foo%09bar' },
+  // newline
+  { path: '/foo\nbar', expected: 'file:///foo%0Abar' },
+  // carriage return
+  { path: '/foo\rbar', expected: 'file:///foo%0Dbar' },
+  // latin1
+  { path: '/fóóbàr', expected: 'file:///f%C3%B3%C3%B3b%C3%A0r' },
+  // Euro sign (BMP code point)
+  { path: '/€', expected: 'file:///%E2%82%AC' },
+  // Rocket emoji (non-BMP code point)
+  { path: '/🚀', expected: 'file:///%F0%9F%9A%80' },
+];
 
-  for (const { path, expected } of testCases) {
-    const actual = url.pathToFileURL(path).href;
-    assert.strictEqual(actual, expected);
-  }
+for (const { path, expected } of windowsTestCases) {
+  const actual = url.pathToFileURL(path, { windows: true }).href;
```

Comment:
A test variation that passes `null` or a non-object value as the `options` would be good.

Reply:
After changing some code, I don't think this test is needed anymore. Any object without the `windows` property (such as a null object) would just get ignored. I don't think it's worth the extra test, but if you disagree, I'll happily implement them anyway.

Reply:
The updated impl is certainly better but I still do think the additional tests are worthwhile.

Reply:
Will do!